yt_dlp/extractor/youtube.py

```
@@ -4930,26 +4930,30 @@ def get_mobj(url):
-                if mobj['tab'] == '/videos' and tab_name.lower() != mobj['tab'][1:]:
-                    redirect_warning = f'The URL does not have a {mobj["tab"][1:]} tab'
-                    if not mobj['not_channel'] and item_id[:2] == 'UC':
-                        # Topic channels don't have /videos. Use the equivalent playlist instead
-                        pl_id = f'UU{item_id[2:]}'
-                        pl_url = f'https://www.youtube.com/playlist?list={pl_id}'
-                        try:
-                            data, ytcfg = self._extract_data(pl_url, pl_id, ytcfg=ytcfg, fatal=True)
-                        except ExtractorError:
-                            redirect_warning += ' and the playlist redirect gave error'
-                        else:
-                            item_id, url, tab_name = pl_id, pl_url, mobj['tab'][1:]
-                            redirect_warning += f'. Redirecting to playlist {pl_id} instead'
-                    if tab_name.lower() != mobj['tab'][1:]:
-                        redirect_warning += f'. {tab_name} tab is being downloaded instead'
+                if selected_tab_name != requested_tab_name and requested_tab_name != 'featured':
+                    redirect_warning = f'The URL does not have a {requested_tab_name} tab'
+                    if requested_tab_name == 'videos':
+                        if not mobj['not_channel'] and item_id[:2] == 'UC':
+                            # Topic channels don't have /videos. Use the equivalent playlist instead
+                            pl_id = f'UU{item_id[2:]}'
+                            pl_url = f'https://www.youtube.com/playlist?list={pl_id}'
+                            try:
+                                data, ytcfg = self._extract_data(pl_url, pl_id, ytcfg=ytcfg, fatal=True)
+                            except ExtractorError:
+                                redirect_warning += ' and the playlist redirect gave error'
```

Hmm I wonder if we should raise an error instead of redirecting to the home tab here?

@pukkandan

---

This should work: https://www.youtube.com/channel/UCtFRv9O2AHqOZjjynzrv-xg